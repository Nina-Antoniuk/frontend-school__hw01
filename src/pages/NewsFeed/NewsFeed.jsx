import React from 'react';
import { v4 as uuid } from 'uuid';
import { PropTypes } from 'prop-types';
import s from './NewsFeed.module.scss';
import Post from '../../components/Post/Post';
import LoaderComponent from '../../components/LoaderComponent/LoaderComponent';
import statuses from '../../consts';
import fetchTrends from '../../services/fetchNews';

const { PENDING, RESOLVE, REJECT, INIT } = statuses;

const NewsFeed = function NewsFeed({ getVideo }) {
  const [trends, setTrends] = React.useState([]);
  const [status, setStatus] = React.useState(INIT);
  const [error, setError] = React.useState('');

  React.useEffect(() => {
    setStatus(PENDING);
    fetchTrends()
      .then((data) => {
        setTrends(data);
        return setStatus(RESOLVE);
      })
      .catch((error_) => {
        setError(error_);
        return setStatus(REJECT);
      });
  }, []);

  React.useEffect(() => {
    if (!trends) return;
    getVideo(trends[0]);
  }, [getVideo, trends]);

  if (status === INIT) {
    return <div />;
  }

  if (status === PENDING) {
    return <LoaderComponent />;
  }

  if (status === REJECT) {
    return (
      <div>
        Oops.. something went wrong!
        {error}
      </div>
    );
  }

  if (status === RESOLVE) {
    return (
      <section className="section">
        {trends ? (
          <ul className={s.postsList}>
            {trends.map((trend) => (
              <Post
                key={uuid()}
                uniqueId={trend.authorMeta.name}
                auth={trend.authorMeta}
                video={trend.videoUrl}
                hashtags={trend.hashtags}
                desc={trend.text}
                views={trend.playCount}
                comments={trend.commentCount}
              />
            ))}
          </ul>
        ) : (
          <p>No data came from the server</p>
        )}
      </section>
    );
  }
};

NewsFeed.propTypes = {
  getVideo: PropTypes.func.isRequired,
};

export default NewsFeed;
