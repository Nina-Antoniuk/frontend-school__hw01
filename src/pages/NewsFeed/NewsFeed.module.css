.postsList {
  max-width: 80%;
  padding-left: 0;
  list-style-type: none;
}

@media screen and (min-width: 768px) {
  .postsList {
    max-width: 70%;
    margin-left: 20%;
  }
}
/*# sourceMappingURL=NewsFeed.module.css.map */
