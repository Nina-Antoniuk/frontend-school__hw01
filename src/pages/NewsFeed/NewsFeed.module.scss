$breakpoint-max-mobile: 767.98px;
$breakpoint-min-tablet: 768px;
$breakpoint-min-desktop: 1200px;

.postsList {
  max-width: 80%;
  padding-left: 0;
  list-style-type: none;

  @media screen and (min-width: $breakpoint-min-tablet) {
    max-width: 70%;
    margin-left: 20%;
  }
}
