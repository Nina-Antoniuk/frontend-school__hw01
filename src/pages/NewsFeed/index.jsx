import React from 'react';
import { v4 as uuid } from 'uuid';
import LoaderComponent from '../../components/LoaderComponent';
import styled from 'styled-components';
import { STATUSES } from '../../shared/consts';
import Post from '../../components/Post';
import { fetchTrends } from '../../api/fetchTrends';
import { size } from '../../shared/css-consts';

const Section = styled.section`
  display: flex;
  justify-content: center;
  padding-top: 30px;

  @media screen and (min-width: ${size.tabletS}) {
    padding-top: 50px;
  }
`;
const PostList = styled.ul`
  max-width: 80%;
  padding-left: 0;
  list-style-type: none;

  @media screen and (min-width: ${size.tabletS}) {
    max-width: 70%;
    margin-left: 20%;
  }
`;
const { PENDING, RESOLVE, REJECT, INIT } = STATUSES;

const NewsFeed = function NewsFeed() {
  const [trends, setTrends] = React.useState([]);
  const [status, setStatus] = React.useState(INIT);
  const [error, setError] = React.useState('');

  React.useEffect(() => {
    setStatus(PENDING);
    fetchTrends()
      .then((data) => {
        setTrends(data);
        return setStatus(RESOLVE);
      })
      .catch((err) => {
        setError(err);
        return setStatus(REJECT);
      });
  }, []);

  if (status === INIT) {
    return <div />;
  }

  if (status === PENDING) {
    return <LoaderComponent />;
  }

  if (status === REJECT) {
    return (
      <div>
        Oops.. something went wrong!
        {error}
      </div>
    );
  }

  if (status === RESOLVE) {
    return (
      <Section>
        {trends ? (
          <PostList>
            {trends.map((trend) => (
              <Post
                key={uuid()}
                video={trend.video.playAddr}
                author={trend.author}
                description={trend.desc}
                stats={trend.stats}
                heart={trend.authorStats.heart}
              />
            ))}
          </PostList>
        ) : (
          <p>No data came from the server</p>
        )}
      </Section>
    );
  }
};

export default NewsFeed;
