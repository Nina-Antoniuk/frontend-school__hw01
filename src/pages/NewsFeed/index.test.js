import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import NewsFeed from './index';

test('Check NewsPages rendering', () => {
  const container = render(<NewsFeed />);
  expect(container).toMatchSnapshot();
});
