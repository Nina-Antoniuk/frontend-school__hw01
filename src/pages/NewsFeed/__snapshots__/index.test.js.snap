// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check NewsPages rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Loader"
      >
        <div
          class="sl-bubble2"
        >
          <div
            class="sl-loader"
            style="--size: 1.50;"
          >
            <div
              class="sl-loader-items"
            >
              <div
                class="sl-bubble"
                style="--i: 1; --bg: #FFFFFF;"
              />
              <div
                class="sl-bubble"
                style="--i: 2; --bg: #FFFFFF;"
              />
              <div
                class="sl-bubble"
                style="--i: 3; --bg: #FFFFFF;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Loader"
    >
      <div
        class="sl-bubble2"
      >
        <div
          class="sl-loader"
          style="--size: 1.50;"
        >
          <div
            class="sl-loader-items"
          >
            <div
              class="sl-bubble"
              style="--i: 1; --bg: #FFFFFF;"
            />
            <div
              class="sl-bubble"
              style="--i: 2; --bg: #FFFFFF;"
            />
            <div
              class="sl-bubble"
              style="--i: 3; --bg: #FFFFFF;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
